src/combinator.rs

```diff
@@ -297,10 +297,16 @@ impl<I: Clone, O1, O2, A: Parser<I, O1, Error = E>, B: Parser<I, O2, Error = E>,
 
 /// See [`Parser::delimited_by`].
 #[derive(Copy, Clone)]
-pub struct DelimitedBy<A, I>(pub(crate) A, pub(crate) I, pub(crate) I);
+pub struct DelimitedBy<A, L, R, U, V> {
+    pub(crate) item: A,
+    pub(crate) start: L,
+    pub(crate) end: R,
+    pub(crate) phantom_start: PhantomData<U>,
+    pub(crate) phantom_end: PhantomData<V>,
```

Comment:
Only a single `PhantomData` field is ever needed: if multiple type parameters are phantoms, you can use `phantom: PhantomData<(U, V)>`